libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -76,21 +76,39 @@
 #define _LIBCUDACXX_CUDACC_BELOW_12_3
 #endif // defined(_LIBCUDACXX_CUDACC) && _LIBCUDACXX_CUDACC_VER < 1203000
 
+// Ensure that a user can disable warnings coming from our headers
+#ifndef _CCCL_IMPLICIT_SYSTEM_HEADER
+#  if defined(_LIBCUDACXX_COMPILER_GCC)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER _Pragma("GCC system_header")
+#  elif defined(_LIBCUDACXX_COMPILER_CLANG)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER _Pragma("clang system_header")
+#  elif defined(_LIBCUDACXX_COMPILER_MSVC)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER __pragma(system_header)
+#  elif defined(_LIBCUDACXX_COMPILER_NVHPC)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER _Pragma("GCC system_header")
+#  else
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER
+#  endif
+#endif // !_CCCL_IMPLICIT_SYSTEM_HEADER
+
 #if defined(_MSC_VER) && !defined(__clang__)
 #  define _LIBCUDACXX_HAS_PRAGMA_MSVC_WARNING
-#  if !defined(_LIBCUDACXX_DISABLE_PRAGMA_MSVC_WARNING)
-#    define _LIBCUDACXX_USE_PRAGMA_MSVC_WARNING
-#  endif
-#else
-#  if !defined(_LIBCUDACXX_DISABLE_PRAGMA_GCC_SYSTEM_HEADER) && !defined(_LIBCUDACXX_COMPILER_NVRTC)
-#    define _LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER
-#  endif
 #endif
 
-#if defined(_LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER)
-#pragma GCC system_header
+#if defined(CCCL_AS_SYSTEM_HEADER)
+#  define _LIBCUDACXX_IMPLICIT_SYSTEM_HEADER
+#elif  defined(_LIBCUDACXX_COMPILER_MSVC) \
```

Comment:
This construction here is needed to keep preexisting behavior. 

I decided to overwrite the old `_LIBCUDACXX_DISABLE_PRAGMA_MSVC_WARNING` and `_LIBCUDACXX_DISABLE_PRAGMA_GCC_SYSTEM_HEADER` behavior in case of `CCCL_AS_SYSTEM_HEADER` 

Shout if you disagree